src/components/ec-file-list/ec-file-list.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div
+    data-test="ec-file-list"
+    class="ec-file-list"
+  >
+    <div
+      v-for="(item, index) in items"
+      :key="item.name"
+      ref="items"
+      class="ec-file-list__item"
+      :data-test="`ec-file-list__item ec-file-list__item--${index}`"
+    >
+
+      <ec-icon
+        class="ec-file-list__doc-icon"
+        data-test="ec-file-list__document-icon"
+        name="simple-document"
+        type="interactive"
+        :size="20"
+      />
+
+      <span
+        class="ec-file-list__name"
+        :title="item.name"
+        :data-test="`ec-file-list__name ec-file-list__name--${index}`"
+      >
+        {{ item.name }}
+      </span>
+
+      <ec-icon
+        class="ec-file-list__delete-icon"
+        :class=" { 'ec-file-list__delete-icon--is-disabled': isDeleteDisabled }"
+        :data-test="`ec-file-list__delete-icon ec-file-list__delete-icon----${index}`"
+        name="simple-outline-delete"
+        type="interactive"
+        :size="20"
+        @click="onDelete(item)"
+      />
+
+    </div>
+  </div>
+</template>
+
+<script>
+import EcIcon from '../ec-icon';
+
+export default {
+  name: 'EcFileList',
+  components: {
+    EcIcon,
+  },
+  props: {
+    items: {
+      type: Array,
+      default: () => [],
+    },
+    isDeleteDisabled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      name: 'test',
+    };
+  },
+  methods: {
+    onDelete(item) {
+      /* istanbul ignore else */
```

Comment:
We can test both of the cases. You can mount the component with `isDeleteDisabled` set to true/false, click on the element and then check what was emitted from the component using this: https://v1.test-utils.vuejs.org/api/wrapper/emitted.html#emitted. For disabled state, the emitted() should return an empty array.

EDIT: I actually found tests like that. Why it needs to be ignored then?

Reply:
We do test both cases (disabled and non disabled emit checks), with the disabled, I check if the emit sends back undefined, the code coverage just didn't want to highlight it? Not sure if I missed something?

Reply:
Do we need to emit the index to delete?

Reply:
I did it this way because the component lists a few items, when we delete we need to know which one the smart component need to remove or have to deal with.